prov/efa/src/efa_mr.c

```diff
@@ -560,6 +556,10 @@ static struct ibv_mr *efa_mr_reg_ibv_mr(struct efa_mr *efa_mr, struct fi_mr_attr
 	 */
 #if HAVE_SYNAPSEAI
 	if (efa_mr_is_synapseai(efa_mr)) {
+		int dmabuf_fd;
```

Comment:
it is libfabric convention that all variables be defined at the top of the function. How should we navigate that rule in this case ... b/c it is messy if you do it that way.  I do think this looks cleaner, but it is the only place in the EFA provider where we are not declaring variables at the top of the function.

Reply:
The convention is not to put variable declaration in the middle of non-declaration statements. For example:
```
f()
{
    my_initializatin();

    struct fi_info *info;

    info = fi_allocinfo();
}
```

It's fine to declare variables at the beginning of a code block (delimitated by `{` and `}`). For the case here, however, if you prefer to keep the declaration at the top, I can surely make the change.

Reply:
A better fix for this issue is to change neuron and synapseai specific symbols into the `ofi_*` function. Then we don't need `HAVE_***` blocks here

Reply:
I will leave it in current shape for now and let your team make improvement in the future. The main purpose of this entire PR is to eliminate all the warning messages so that `-Werror` flag can be added.